components/httpclient/httpclient.go

```diff
@@ -38,15 +38,18 @@ func (p *HttpClient) NewRequest() *sling.Sling {
 func (p *HttpClient) DoRequest(s *sling.Sling, res interface{}) error {
 	var serverError ServerError
 	resp, err := s.Receive(res, &serverError)
-	if err == nil {
-		defer resp.Body.Close()
-		if !(200 <= resp.StatusCode && resp.StatusCode < 300) {
-			err = serverError
-		} else if res != nil {
-			if reflect.TypeOf(res).Kind() == reflect.Struct {
-				err = p.validate.Struct(res)
-			}
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if !(200 <= resp.StatusCode && resp.StatusCode < 300) {
+		return serverError
+	}
+	if res != nil {
+		rv := reflect.ValueOf(res)
+		if rv.Kind() == reflect.Ptr && rv.Elem().Kind() == reflect.Struct {
```

Comment:
This will fail when the struct is not passed as a reference. Examaple: `httpClient.DoRequest(httpClient.NewRequest().Path("good").Get(""), res2)` instead of `httpClient.DoRequest(httpClient.NewRequest().Path("good").Get(""), &res2)`.

It make sense to use this with a reference, but the signatue of the function allow both things. I thing you could keep the old if statement: `if reflect.TypeOf(res).Kind() == reflect.Struct {`.

Reply:
Passing non nil without reference should be an error.  There's no way you can get the result outside the function if you don't pass a reference.  If you pass a struct without reference you are making a copy of the struct inside this function that disappears once the function ends.  `encoding/json` behaves the same way, check this out: https://play.golang.org
/p/bgdVUsrkEaW so I think this behavior is correct.

Since go doesn't have generics, any time an `interface{}` is used it is expected that you are bypassing the type system of the compiler and relaying on runtime checks, so improper types may result in runtime errors.